core/lib/resources/presence/presence_manager.js

```diff
@@ -359,9 +375,16 @@ PresenceManager.prototype.getClientsOnline = function() {
   var result = {};
 
   function processMessage(message) {
-    result[message.userId] = result[message.userId] ||
-                { clients: { } , userType: message.userType };
-    result[message.userId].clients[message.clientId] = message.userData || {};
+    result[message.userId] = result[message.userId] || { 
+      clients: { },
+      userType: message.userType
+    };
+    
+    var payload = underscore.extend({}, 
+                                    (message.userData || {}), 
+                                    (message.clientData || {}));
```

Comment:
For now, this is fine (Because we have to deploy server first.)

But make a note to add a cleanup to this. The server should eventually handle this properly. We should change the client to take care of this properly by parsing the message and extending etc. Once we are sure there is nothing terrible that will happen, we can change the format of get/sync and make a minor version update.